qiskit_metal/draw/utility.py

```diff
@@ -609,3 +610,250 @@ def snap_unit_vector(vec_n: Vec2D, flip: bool = False) -> Vec2D:
+
+        Returns:
+            np.ndarray: return a new np.ndarray object
+        """
+        return Vec3D.add(vec, translate_vec)
+
+    @staticmethod
+    def rotate(vec: np.ndarray,
+               cvec: np.ndarray,
+               ax: bool = False,
+               ay: bool = False,
+               az: bool = False,
+               radians: float = 0.0):
+        """Rotate a vector
+
+        Args:
+            vec (np.ndarray): vector to transform
+            cvec (np.ndarray): center of rotation as a vector
+            ax (bool, optional): Rotate around x-axis. Defaults to False.
+            ay (bool, optional): Rotate around y-axis. Defaults to False.
+            az (bool, optional): Rotate around z-axis. Defaults to False.
+            radians (float, optional): radians of rotation. Defaults to 0.0.
+
+        Returns:
+            np.ndarray: return a new np.ndarray object
+        """
+        if not isinstance(vec, np.ndarray):
+            vec = np.array(vec)
+        if not isinstance(cvec, np.ndarray):
+            cvec = np.array(cvec)
+
+        if ax:
+            rot_x = np.array([[1, 0, 0], [0,
+                                          np.cos(radians), -np.sin(radians)],
+                              [0, np.sin(radians),
+                               np.cos(radians)]])
+            translate_vec = np.array([0, cvec[1], cvec[2]])
+            new_vec = rot_x.dot((vec - translate_vec)) + translate_vec
+
+        if ay:
+            rot_y = np.array([[np.cos(radians), 0,
+                               np.sin(radians)], [0, 1, 0],
+                              [-np.sin(radians), 0,
+                               np.cos(radians)]])
+            translate_vec = np.array([cvec[0], 0, cvec[2]])
+            new_vec = rot_y.dot((vec - translate_vec)) + translate_vec
+
+        if az:
+            rot_z = np.array([[np.cos(radians), -np.sin(radians), 0],
+                              [np.sin(radians),
+                               np.cos(radians), 0], [0, 0, 1]])
+            translate_vec = np.array([cvec[0], cvec[1], 0])
+            new_vec = rot_z.dot((vec - translate_vec)) + translate_vec
+
+        return new_vec
+
+    @staticmethod
+    def angle_elevation(vec: np.ndarray) -> float:
+        """Returns the elevation angle of vector
+
+        Args:
+            vec (np.ndarray): input vector
+
+        Returns:
+            float: elevation angle in radians
+        """
+        unit_z = np.array([0, 0, 1])
+        unit_self = Vec3D.normed(vec)
+        return np.arccos(unit_self.dot(unit_z))
+
+    @staticmethod
+    def angle_azimuth(vec: np.ndarray) -> float:
+        """Returns the azimuthal angle of vector
+
+        Args:
+            vec (np.ndarray): input vector
+
+        Returns:
+            float: azimuthal angle of vector
+        """
+        return np.arctan2(*vec[:2][::-1])
+
+    @staticmethod
+    def two_points_described(points2D: List[np.ndarray]) -> Tuple[np.ndarray]:
```

Comment:
Why would we not want to describe two points in 3D?

Reply:
Describing two points in 3D i.e. finding distance and unit vector can be done. But a 3D vector will have infinitely many tanget vectors lying on the plane described the normal vector as our given vector. This will create ambiguity in the result, and hence this wasn't implemented. Any suggestions as to how this can be resolved are welcome!